### DISCUSSION TITLE
Tutorial deploy failed on netlify

### DISCUSSION THREAD

**Discussion by arsser** [refinedev/refine#4351] [Q&A] [answered]
Toturial deploy failed on netlify，any help？

6:57:49 PM: $ npm run build
6:57:49 PM: > tutorial@0.1.0 build
6:57:49 PM: > refine build
6:57:53 PM: Creating an optimized production build...
6:58:09 PM: Failed to compile.
6:58:09 PM: 
6:58:09 PM: Module not found: Error: Can't resolve '@refinedev/react-table' in '/opt/build/repo/node_modules/@refinedev/inferencer/dist/esm'
6:58:09 PM: ​
6:58:09 PM: "build.command" failed                                        
6:58:09 PM: ────────────────────────────────────────────────────────────────
6:58:09 PM: ​
6:58:09 PM:   Error message
6:58:09 PM:   Command failed with exit code 1: npm run build (https://ntl.fyi/exit-code-1)
6:58:09 PM: ​
6:58:09 PM:   Error location
6:58:09 PM:   In Build command from Netlify app:
6:58:09 PM:   npm run build
6:58:09 PM: ​
6:58:09 PM:   Resolved config
6:58:09 PM:   build:
6:58:09 PM:     command: npm run build
6:58:09 PM:     commandOrigin: ui
6:58:09 PM:     publish: /opt/build/repo/build
6:58:09 PM:     publishOrigin: ui
6:58:11 PM: Failed during stage 'building site': Build script returned non-zero exit code: 2 (https://ntl.fyi/exit-code-2)
6:58:10 PM: Build failed due to a user error: Build script returned non-zero exit code: 2
6:58:11 PM: Failing build: Failed to build site
6:58:11 PM: Finished processing build request in 1m0.912s

**Comment by aliemir** [comment]
Hey @arsser, which UI package you've selected in tutorial? Let's try to investigate the cause of the problem.

**Reply by aliemir** [reply, marked as answer]
For the headless option, you'll also need to install `@refinedev/react-table` and `@refinedev/react-hook-form`, can you confirm these packages are installed? 🙏

**Reply by arsser** [reply]
yes, It worked!Thanks so much. As a beginner, I'm curious as to why it was working normally when I ran it locally before, seemingly without individually installing these two packages.
